packages/eslint-plugin-react-hooks/__tests__/ESLintRulesOfHooks-test.js

```diff
@@ -349,6 +349,30 @@ const tests = {
     `,
   ],
   invalid: [
+    {
+      code: `
+      // Invalid because it's dangerous and might not warn otherwise.
+      // This *must* be invalid.
+      const Component = wrapperFunction(() => {
+        if (cond) {
+          useConditionalHook();
+        }
```

Comment:
I know the current heuristic doesn't check for `return` to tell whether something is a component or not, but the absence of a `return` here makes it look suspicious to me as the one non-symbol value components are not allowed to return is `undefined`.

Reply:
Should I just return null at the end of the component? To be honest I did not include a return because none of the other tests did.